lib/structures/GuildTextableChannel.js

```diff
@@ -0,0 +1,213 @@
+    }
+
+    /**
+     * Get a list of users who reacted with a specific reaction
+     * @arg {String} messageID The ID of the message
+     * @arg {String} reaction The reaction (Unicode string if Unicode emoji, `emojiName:emojiID` if custom emoji)
+     * @arg {Object} [options] Options for the request. If this is a number, it is treated as `options.limit` ([DEPRECATED] behavior)
+     * @arg {String} [options.after] Get users after this user ID
+     * @arg {Number} [options.limit=100] The maximum number of users to get
+     * @arg {String} [before] [DEPRECATED] Get users before this user ID. Discord no longer supports this parameter
+     * @arg {String} [after] [DEPRECATED] Get users after this user ID
+     * @returns {Promise<Array<User>>}
+     */
+    getMessageReaction(messageID, reaction, options, before, after) {
+        return this._client.getMessageReaction.call(this._client, this.id, messageID, reaction, options, before, after);
+    }
+
+    /**
+     * Get previous messages in the channel
+     * @arg {Object} [options] Options for the request. If this is a number ([DEPRECATED] behavior), it is treated as `options.limit`
+     * @arg {String} [options.after] Get messages after this message ID
+     * @arg {String} [options.around] Get messages around this message ID (does not work with limit > 100)
+     * @arg {String} [options.before] Get messages before this message ID
+     * @arg {Number} [options.limit=50] The max number of messages to get
+     * @arg {String} [before] [DEPRECATED] Get messages before this message ID
+     * @arg {String} [after] [DEPRECATED] Get messages after this message ID
+     * @arg {String} [around] [DEPRECATED] Get messages around this message ID (does not work with limit > 100)
+     * @returns {Promise<Array<Message>>}
+     */
+    getMessages(options, before, after, around) {
+        return this._client.getMessages.call(this._client, this.id, options, before, after, around);
+    }
+
+    /**
+     * Purge previous messages in the channel with an optional filter
+     * @arg {Object} options Options for the request. If this is a number ([DEPRECATED] behavior), it is treated as `options.limit`
+     * @arg {String} [options.after] Get messages after this message ID
+     * @arg {String} [options.before] Get messages before this message ID
+     * @arg {Function} [options.filter] Optional filter function that returns a boolean when passed a Message object
+     * @arg {Number} options.limit The max number of messages to search through, -1 for no limit
+     * @arg {String} [options.reason] The reason to be displayed in audit logs
+     * @arg {Function} [filter] [DEPRECATED] Optional filter function that returns a boolean when passed a Message object
+     * @arg {String} [before] [DEPRECATED] Get messages before this message ID
+     * @arg {String} [after] [DEPRECATED] Get messages after this message ID
+     * @arg {String} [reason] [DEPRECATED] The reason to be displayed in audit logs
```

Comment:
Time to remove all of these deprecated opts? They've been deprecated for like 17 years, and possibly unpopular opinion but anyone who has yet to change a single line of code to support the new way of doing it should get a new hobby

Reply:
If I get round to sorting out these deprecations, they'll be in a separate PR. This PR is a little complicated as it is already